adaptdl/adaptdl/torch/_metrics.py

```diff
@@ -45,25 +41,31 @@ def profile_sync_time(sync_time):
 _PREV_REPORT = None
 
 
-def profile_step_commit():
+def profile_step_commit(accumulation_step=False):
     global _PREV_REPORT
     state = _metrics_state()
     step_time = time.time() - state.step_start
     num_nodes = adaptdl.env.num_nodes()
     num_replicas = adaptdl.env.num_replicas()
-    key = (num_nodes, num_replicas, state.local_bsz)
-    state.profile[key]["step_time"] += step_time
-    state.profile[key]["sync_time"] += state.sync_time
-    state.profile[key]["count"] += 1
+    key = (num_nodes, num_replicas,
+           state.local_bsz, state.gradient_accumulation_steps)
+    if accumulation_step:
+        state.profile[key]["accumulation_step_time"] += step_time
+    else:
+        state.profile[key]["step_time"] += step_time
+        state.profile[key]["sync_time"] += state.sync_time
+        state.profile[key]["count"] += 1
```

Comment:
I think there's a chance of mis-counting if there's a checkpoint-restart during an accumulation step?

Reply:
Fixed